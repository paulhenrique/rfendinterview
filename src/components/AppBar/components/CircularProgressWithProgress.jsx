import { Box, CircularProgress, Typography } from "@mui/material";
import { green } from "@mui/material/colors";
import React from "react";
import { getPercent } from "@/pages/Score/components/QuizCard";

const CircularProgressWithProgress = ({ score = 0, total = 0 }) => {
  return (
    <Box
      sx={{
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      <CircularProgress
        variant="determinate"
        size={80}
        sx={{ color: green["600"] }}
        value={getPercent(score, total)}
      />
      <Typography sx={{ position: "absolute" }}>
        {`${score}/${total}`}
      </Typography>
    </Box>
  );
};

export default CircularProgressWithProgress;
