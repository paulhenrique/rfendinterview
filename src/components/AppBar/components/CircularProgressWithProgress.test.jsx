/**
 * @jest-environment jsdom
 */
import React from "react";
import CircularProgressWithProgress from "./CircularProgressWithProgress";
import { render, screen } from "@testing-library/react";

describe("Circular Progress With Progress", () => {
  it("should render correctly", () => {
    const { container } = render(<CircularProgressWithProgress value={50} />);
    expect(container).toMatchSnapshot();
  });

  it("should value passed to circular progress to be in screen", async () => {
    const value = 50;
    render(
      <CircularProgressWithProgress value={value} score={value} total={value} />
    );
    expect(await screen.findByText(`${value}/${value}`));
  });
});
