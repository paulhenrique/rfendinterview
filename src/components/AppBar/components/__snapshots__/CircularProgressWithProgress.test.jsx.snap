// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circular Progress With Progress > should render correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1n2mv2k"
  >
    <span
      aria-valuenow="NaN"
      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-kmkkre-MuiCircularProgress-root"
      role="progressbar"
      style="width: 80px; height: 80px; transform: rotate(-90deg);"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
          style="stroke-dasharray: 126.920; stroke-dashoffset: NaNpx;"
        />
      </svg>
    </span>
    <p
      class="MuiTypography-root MuiTypography-body1 css-2atoo-MuiTypography-root"
    >
      0/0
    </p>
  </div>
</div>
`;

exports[`Circular Progress With Progress should render correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1n2mv2k"
  >
    <span
      aria-valuenow="NaN"
      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-kmkkre-MuiCircularProgress-root"
      role="progressbar"
      style="width: 80px; height: 80px; transform: rotate(-90deg);"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
          style="stroke-dasharray: 126.920; stroke-dashoffset: NaNpx;"
        />
      </svg>
    </span>
    <p
      class="MuiTypography-root MuiTypography-body1 css-2atoo-MuiTypography-root"
    >
      0/0
    </p>
  </div>
</div>
`;
